src/main/java/com/google/api/codegen/metadatagen/PackageMetadataContext.java

```diff
@@ -0,0 +1,124 @@
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+import org.yaml.snakeyaml.Yaml;
+
+/*
+ * A Context object to be passed into the snippet template rendering phase of package metadata
+ * generation.
+ */
+public class PackageMetadataContext {
+  private Map<String, Object> dependenciesMap = new HashMap<>();
+
+  private Map<String, Object> defaultsMap = new HashMap<>();
+
+  private Doc copierResults;
+
+  private ApiNameInfo apiNameInfo;
+
+  /**
+   * Constructor.
+   *
+   * @param apiNameInfo Represents naming information for the given API.
+   * @param copierResults Results from the copier phase of package metadata generation.
+   */
+  public PackageMetadataContext(ApiNameInfo apiNameInfo, Doc copierResults) {
+    this.apiNameInfo = apiNameInfo;
+    this.copierResults = copierResults;
+  }
+
+  /**
+   * Initializes package metadata configuration. Should be called before passing this context to a
+   * snippet template environment.
+   *
+   * @param dependenciesFile The path to the dependencies file
+   * @param defaultsFile The path to the defaults file
+   * @throws IOException
+   */
+  @SuppressWarnings("unchecked")
+  public void loadConfigurationFromFile(String dependenciesFile, String defaultsFile)
+      throws IOException {
+    Yaml yaml = new Yaml();
+
+    String dependencies =
+        new String(Files.readAllBytes(Paths.get(dependenciesFile)), StandardCharsets.UTF_8);
+    this.dependenciesMap = (Map<String, Object>) yaml.load(dependencies);
+
+    String defaults =
+        new String(Files.readAllBytes(Paths.get(defaultsFile)), StandardCharsets.UTF_8);
+    this.defaultsMap = (Map<String, Object>) yaml.load(defaults);
+  }
+
+  public String getDependencyValue(String keySpecifier) {
+    return getUncheckedMapValue(dependenciesMap, keySpecifier);
+  }
+
+  public String getDefaultsValue(String keySpecifier) {
+    return getUncheckedMapValue(defaultsMap, keySpecifier);
+  }
+
+  @SuppressWarnings("unchecked")
+  // Note: Snippets doesn't seem to support variadic args, so we use ":" as a separator.
+  //
+  // TODO (geigerj): this is a hack to port the dynamic Javascript configuration from Packman to
+  // Java. To hew closer to Java style, investigate defining configuration information in a
+  // well-defined format that can be converted properly to Java types.
+  private String getUncheckedMapValue(Map<String, Object> map, String keySpecifier) {
+    String[] keys = keySpecifier.split(":");
+    Map<String, Object> current = map;
+    for (int i = 0; i < keys.length - 1; i++) {
+      current = (Map<String, Object>) current.get(keys[i]);
+    }
+    return (String) current.get(keys[keys.length - 1]);
+  }
+
+  public ApiNameInfo getApiNameInfo() {
+    return apiNameInfo;
+  }
+
+  // TODO (geigerj): accessing apiNameInfo directly in the snippets leads to errors, hence these
+  // aliases in the context. Investigate whether these are truly needed.
```

Comment:
What kind of errors? Is it due to the fact that the methods in ApiNameInfo are package-private?